pkg/plugins/golang/v2/scaffolds/api.go

```diff
@@ -89,11 +90,14 @@ func (s *apiScaffolder) scaffold() error {
 	doAPI := s.resource.HasAPI()
 	doController := s.resource.HasController()
 
-	if doAPI {
-
+	// config v2 should only be updated if the API is scaffolded
+	if doAPI || s.config.GetVersion().Compare(cfgv2.Version) == 1 {
 		if err := s.config.UpdateResource(s.resource); err != nil {
```

Comment:
We should update the resource always not only is doAPI. 
If we create a webhook, or only a controller or an api, I mean in any scenario we need to store the info.